sql/sys_vars.cc

```diff
@@ -6683,3 +6683,28 @@ static Sys_var_ulonglong Sys_max_rowid_filter_size(
        SESSION_VAR(max_rowid_filter_size), CMD_LINE(REQUIRED_ARG),
        VALID_RANGE(1024, (ulonglong)~(intptr)0), DEFAULT(128*1024),
        BLOCK_SIZE(1));
+
+static Sys_var_mybool Sys_redirect_enabled(
+       "redirect_enabled",
+       "Enable return redirection endpoint",
+       GLOBAL_VAR(redirect_enabled), CMD_LINE(OPT_ARG),
+       DEFAULT(FALSE), NO_MUTEX_GUARD, NOT_IN_BINLOG, ON_CHECK(0),
+       ON_UPDATE(0));
+
+static Sys_var_charptr Sys_redirect_server_host(
+       "redirect_server_host",
+       "The server host that client can choose to redirect to.",
+       READ_ONLY GLOBAL_VAR(redirect_server_host), CMD_LINE(OPT_ARG),
+       DEFAULT("localhost"));
```

Comment:
Do these defaults mean that the server redirects to itself by default? I think this might be a case where a default value isn't that valuable.

Reply:
Yes, it does. The purpose is to avoid the cases that cause connection failure, where redirection is enabled on both server and client, but redirection info is not provided appropriately. Just wanna play prudently here... If you believe such protection is unneccesary or should be conducted elsewhere, I can remove the default values.